### ISSUE TITLE
Automatically create a local tool-manifest when a tool is installed and no manifest exists

### ISSUE THREAD

**Issue by baronfel** [dotnet/sdk#27736]
### Is your feature request related to a problem? Please describe.
Right now when a user installs a tool into a new project, they immediately get an error instead of a successful tool installation

```shell
 ➜ dotnet tool install fantomas --prerelease
Cannot find a manifest file.
For a list of locations searched, specify the "-d" option before the tool name.
If you intended to install a global tool, add `--global` to the command.
If you would like to create a manifest, use `dotnet new tool-manifest`, usually in the repo root directory.
```

### Describe the solution you'd like
When installing a local tool, if no manifest exists in the heirarchy one should be created inside a .config directory at the current working directory. There's no need for a user to execute an entirely different command (template creation of a tool manifest) to get unblocked here. We should log that this action was done on the user's behalf. If the user doesn't want this behavior, they can remove + recreate the file at the desired level of the heirarchy.

**Comment by tboby** [comment]
There is some more discussion about this here: #15254

It'd be great to have though! Having to litter `dotnet new tool-manifest` all over my CI/CD just to run post-processing tools like test results converters is a pain. And global tools don't work for shared runners.

**Comment by augustoproiete** [comment]
This seems like a duplicate of *[When installing a .NET local tool, create the tool manifest if one does not exist
#15254](https://github.com/dotnet/sdk/issues/15254)*

Related comment:

> [augustoproiete](https://github.com/augustoproiete) commented [on May 17, 2021](https://github.com/dotnet/sdk/issues/10310#issuecomment-842686168)
> 
> .NET tools is lacking the ability to have a simple command that does exactly what it's being asked: Ensure that a .NET tool is installed and exit gracefully.
> 
> The current state leave us with poor a UX:
> 
> ![image](https://user-images.githubusercontent.com/177608/118564379-8b718b00-b746-11eb-95e6-4b06eb5541cf.png) https://www.nuget.org/packages/dotnet-reportgenerator-globaltool/
> 
> You should be able to copy/paste the instructions on how to install a .NET tool, and it should work on any machine regardless if the tool is installed or not, manifest or not... But that's not where we are today.

---

It's worth noting that creating the manifest automatically would be a great step forward, when the manifest already exists *and* the tool is already installed, the command fails... [We need it to not fail (i.e. error code == 0) if the tool we want already exists](https://github.com/dotnet/sdk/issues/10310).

Also, [if you have a `global.json` it gets in the way of installing tools as well](https://github.com/dotnet/sdk/issues/14721)

**Comment by baronfel** [comment]
You are both correct, I'm going to close this in favor of #15254 and add that one to our candidate list for improvements to tools in .NET 8.
